### PR TITLE
Patch UCX to add CUDA support at runtime

### PR DESCRIPTION
**Pull request by Micket**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by Micket** [comment]
So, I think this works. ucx_info manages to find and report on CUDA stuff.

But, there are some details to sort out when it comes to building the additional UCX-plugins; I would like them link to the existing libuct.so etc. libraries. Right now, I hackishly install the plugins. Alternatively, we can just install copies of those libraries, they should hopefully be identical anyway.

**Comment by boegel** [comment]
@Micket Do you think it's worth implementing a custom easyblock for this, where we can check the output produced by `ucx_info`a bit more thoroughly?

This easyconfig looks complex enough to warrant that, I think...

**Comment by Micket** [comment]
Sure, though it's mostly just the same as UCX.  We could probably automatically populate EB_xxx_MODULES here.

The biggest issue is 
1. Gotta test to see if it actually works for transferring data. I need 2 nodes to test and my cluster is a bit booked up at the moment.
2. The hacky way I install just the modules right now is not good. I would really need to rework this. Maybe even requiring a patch to just build the plugins and not involve any other linking shenanigans. I'm not to familiar with how to hack this into libtool/autohell stuff going on here. I really hate autotools...

File: easybuild/easyconfigs/u/UCX/UCX-1.10.0-dynamic_modules.patch [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+     char *module_name;
++    char *eb_modules;
++    char env[50];
+ 
+     ucs_module_loader_init_paths();
+ 
+     UCS_INIT_ONCE(init_once) {
+         ucs_module_debug("loading modules for %s", framework);
+-        modules_str = ucs_strdup(modules, "modules_list");
++        sprintf(env, "EB_%s_MODULES", framework);
```
Comment (akesandgren) [inline easybuild/easyconfigs/u/UCX/UCX-1.10.0-dynamic_modules.patch:27]: Use strncpy, strncat, strncat here instead, then you can drop stdio and stdlib and do string.h instead.

**Review by akesandgren — CHANGES_REQUESTED** [review]

**Comment by akesandgren** [comment]
This one no longer need the original UCX nor the patch for dynamic modules.

**Comment by Micket** [comment]
Replaced with #13260

**Pull request closed without merging**